src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/SqlConfigurableRetryLogicManager.CreateProvider.cs

```diff
@@ -0,0 +1,334 @@
+            }
+
+            Type type = null;
+            try
+            {
+                // Resolve an Type object from the given type name
+                // Different implementation in .NET Framework & .NET Core
+                type = LoadType(configurableRetryType);
+            }
+            catch (Exception e)
+            {
+                // Try to use the 'SqlConfigurableRetryFactory' as a default type to discover the retry methods
+                // if there was any problem to resolve the 'configurableRetryType' type.
+                type = typeof(SqlConfigurableRetryFactory);
+                SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Unable to load the '{2}' type; Tries to use the internal `{3}` type: {4}",
+                                                       s_typeName, methodName, configurableRetryType, type.FullName, e);
+            }
+
+            // Run the function by the resolved values to get the SqlRetryLogicBaseProvider object
+            try
+            {
+                // Create an instance from the discovered type by its default cunstructor
+                object result = CreateInstance(type, retryMethod, option);
+
+                if (result is SqlRetryLogicBaseProvider provider)
+                {
+                    SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> The created instace is a {2} type.",
+                                                           s_typeName, methodName, typeof(SqlRetryLogicBaseProvider).FullName);
+                    provider.Retrying += OnRetryingEvent;
+                    return provider;
+                }
+            }
+            catch (Exception e)
+            {
+                // Openning a connection or executing a command inside this function 
+                // runs the application to the `TargetInvocationException`.
+                // And using an isolated zone like a specific AppDomain makes an infinit loop.
+                throw new Exception($"Exception is occured to run the `{type.FullName}.{retryMethod}()` method.", e);
+            }
+
+            SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Unable to resolve a valid provider; Returns `null`.", s_typeName, methodName);
+            return null;
+        }
+
+        private static object CreateInstance(Type type, string retryMethodName, SqlRetryLogicOption option)
+        {
+            string methodName = MethodBase.GetCurrentMethod().Name;
+            SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Entry point.", s_typeName, methodName);
+
+            if (type == typeof(SqlConfigurableRetryFactory) || type == null)
+            {
+                SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> The given type `{2}` infers as internal `{3}` type."
+                                                      , s_typeName, methodName, type?.Name ?? NullConst, typeof(SqlConfigurableRetryFactory).FullName);
```

Comment:
This null handling is already done inside TryTraceEvent, need not handle this here.
NullConst can be avoided.